skopt/optimizer/base.py

```diff
@@ -303,6 +312,23 @@ def base_minimize(func, dimensions, base_estimator,
         next_x = np.clip(
             next_x, transformed_bounds[:, 0], transformed_bounds[:, 1])
         next_x = space.inverse_transform(next_x.reshape((1, -1)))[0]
+
+        if len(cat_inds) == 0:
+            if np.any(np.apply_along_axis(lambda x: np.allclose(x, next_x), 1, Xi)):
+                warnings.warn(
+                    "candidate point at the end of %d model fit, already chosen" % i)
+        else:
+            next_x_arr = np.array(next_x)
+            for x in Xi:
+                x_arr = np.array(x)
+                cat_eq = np.all(x_arr[cat_inds] == next_x_arr[cat_inds])
+                non_cat_eq = np.allclose(
+                    np.array(x_arr[non_cat_inds], dtype=np.float32),
+                    np.array(next_x_arr[non_cat_inds], dtype=np.float32))
+                if cat_eq and non_cat_eq:
+                    warnings.warn(
+                        "candidate point at the end of %d model fit, already chosen" % i)
```

Comment:
Okay, never mind, just read the discussion in #170 again, where we assumed that by default the function should be considered stochastic, hence reevaluating it might be worth something.